src/Factory.php

```diff
@@ -140,7 +140,7 @@ function ($_, $reject) use ($promise) {
                 $reject(new \RuntimeException('Cancelled creating socket during DNS lookup'));
 
                 // (try to) cancel pending DNS lookup, otherwise ignoring its results
-                if ($promise instanceof CancellablePromiseInterface) {
+                if (method_exists($promise, 'cancel')) {
```

Comment:
Nit: Suggested as per your #28:

```suggestion
                if (\method_exists($promise, 'cancel')) {
```

It also looks like the now unneeded import for `CancellablePromiseInterface` is still in place.

Reply:
My bad, thought I removed that :+1: